Automattic/sensei
# Add modal window for email signup

This is based on "Version 4" of the designs here: p4CBkD-JP-p2

Before merging, at the very least we need to do the following:

- [x] Only show the modal on initial activation.
- [x] Update the code to push to our production mailing list, not the test list.

## Testing instructions

- Switch to this branch and load any page in Sensei.
- The modal window should pop up.
- Before testing various cases, reload the page and ensure the modal does not pop up the second time, regardless of whether it was submitted. Also, deactivate and reactivate Sensei, and ensure the modal does not appear.
- To see the modal again, deactivate Sensei, delete the option `sensei_installed` from the `wp_options` table, and reactivate Sensei.
- In the modal signup, try entering a valid email, invalid email, no email, the same email that was already submitted, etc.
- Note that this is submitting email addresses to the production email list, so feel free to delete them (or ask someone to do so) after you are done testing.

## Things to discuss (Edit: these are discussed below)

- The design should be close to the P2 post, but feel free to suggest tweaks.
- I'm using a MailChimp form that behaves in the default way, which is to open the MailChimp page in a new tab. This seemed to be the simplest way to do it.
  - Are we ok with this functionality?
  - If so, we should confirm that the information on that page is correct, and the links/buttons go where we want them to.
- I feel that it would be better for this form to be hosted outside of the plugin somehow (perhaps on SenseiLMS.com) and then displayed either using JS or an iframe. This will add some development time, but may be a good long-term decision. That said, the benefit may not be huge (this form will only be shown on initial activation, so having the flexibility to change it quickly may not be very helpful).
